src/os/os_chmod_test.go

```diff
@@ -28,4 +30,26 @@ func TestChmod(t *testing.T) {
 		t.Fatalf("chmod %s %#o: %s", f.Name(), fm, err)
 	}
 	checkMode(t, f.Name(), fm)
+
+}
+
+func TestChown(t *testing.T) {
+	f := newFile("TestChown", t)
+	defer Remove(f.Name())
+	defer f.Close()
+
+	if runtime.GOOS != "windows" {
+		if err := Chown(f.Name(), 0, 0); err != nil {
+			t.Fatalf("chown %s 0 0: %s", f.Name(), err)
+		}
+
+		fi, err := Stat(f.Name())
+		if err != nil {
+			t.Fatalf("stat %s: %s", f.Name(), err)
+		}
+
+		if fi.Sys().(*syscall.Stat_t).Uid != 0 || fi.Sys().(*syscall.Stat_t).Gid != 0 {
```

Comment:
I've had tests explode with complex casts like this. A more common pattern woudl be
if fi.Sys() == nil {
error
}
s, ok := fi.Sys().(*syscall.Stat_t)
if  ! ok {
bad
}
uid, gid = s.Uid, s.Gid // just useful so you're not repeating fi.Sys().(*syscall.Stat_t). everywhere